core/tests/test_utils.py

```diff
@@ -622,6 +622,9 @@ def process_and_flush_pending_tasks(self, queue_name=None):
                 if task.url == '/_ah/queue/deferred':
                     from google.appengine.ext import deferred
                     deferred.run(task.payload)
+                elif feconf.EMAILS_TASK_PREFIX in task.url:
+                    # Flush task if task is of emails taskqueue.
```

Comment:
Oh, when you say user_id, do you mean the params/payload that is passed?

Well ... MapReduce also passes complex payloads, I believe. You may have to do a json.dumps() on the dict in order to convert it to a string, and reconstitute it at the other end. Would that help? We need to find a way to get around the test skew.

Also, I looked [here](https://cloud.google.com/appengine/docs/python/taskqueue/tasks#constructor_1) and noticed this sentence, under 'params':
```
    Params are encoded as application/x-www-form-urlencoded and set to the payload.
```
Do you know what this means? I'm just asking because it might mean that we can retrieve the params as part of the payload after all, but I'm not sure...